packages/webviz-core/src/panels/Teleop/index.js

```diff
@@ -0,0 +1,291 @@
+    };
+    const [x, y, z, th] = COMMANDS[key];
+    const speed = 0.5, turn = 1.0;
+    return {
+      linear: {
+        x: x*speed,
+        y: y*speed,
+        z: z*speed
+      },
+      angular: {
+        x: 0,
+        y: 0,
+        z: th * turn
+      }
+    }
+  };
+
+  _onChange = (event: SyntheticInputEvent<HTMLTextAreaElement>) => {
+    this.props.saveConfig({ value: event.target.value });
+  };
+
+  _onCommandButtonClick = (command) => {
+    // Simulate key press and release
+    this._handleKey({key: command, type: "keydown"});
+    this._handleKey({key: command, type: "keyup"});
+  };
+
+  _handleKey = (event) => {
+    this.setState(function(state, props) {
+      let newPressing = state.pressing;
+      newPressing[event.key] = event.type == "keydown";
+      if(this._publisher.current){
+        if(event.type=="keydown" ){
+          this._publisher.current.publish(this.composeTwist(event.key));
+        }
+      } else {
+        console.error("Publisher not set!");
+      }
+      return {
+        pressing: newPressing
+      };
+    });
+
+    this.forceUpdate(); // HACK Appears to be required to update the colors quickly???
+  };
+
+  // Add handler here for all desired keybindings (they are all the same!)
+  _keyHandlers = {
+    ArrowLeft: this._handleKey,
+    ArrowRight: this._handleKey,
+    ArrowUp: this._handleKey,
+    ArrowDown: this._handleKey,
+    " ": this._handleKey, // Space
+  };
+
+  _renderMenuContent() {
+    const { config, saveConfig } = this.props;
+
+    return (
+      <>
+        <Item>
+          <PanelToolbarLabel>Button color (rgba or hex)</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.buttonColor}
+            onChange={(event) => {
+              saveConfig({ buttonColor: event.target.value });
+            }}
+            placeholder="rgba(1,1,1,1) or #FFFFFF"
+          />
+        </Item>
+        <Item>
+          <PanelToolbarLabel>Topic name</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.topicName}
+            onChange={(event) => {
+              saveConfig({ topicName: event.target.value });
+            }}
+            placeholder="Choose /cmd_vel if unsure"
+          />
+        </Item>
+      </>
+    );
+  }
+
+  render() {
+    const {
+      capabilities,
+      topics,
+      config: { buttonColor, topicName },
+    } = this.props;
+
+    const { datatypeNames, parsedObject, error, pressing } = this.state;
+    const canPublish = capabilities.includes(PlayerCapabilities.advertise);
+
+    return (
+      <Flex col style={{ height: "100%", padding: "12px" }} onKeyPress={this.handleKey}>
+        {topicName && (
+          <Publisher ref={this._publisher} name="Publish" topic={topicName} datatype="geometry_msgs/Twist" />
+        )}
+        
+        <KeyListener global keyDownHandlers={this._keyHandlers} keyUpHandlers={this._keyHandlers} />
```

Comment:
Let's keep the key listerner local? We'd want to preserve the arrow keys for global usage, e.g. currently left/right arrow will seek back/forward 100ms.

Reply:
Done. It makes things a bit awkward, though, you now need to click on any button before the key presses are sent to the component. I guess it can't be avoided, you need to give focus to the component if it's not global.